### PR TITLE
Some minimal details in the response for `POST /namespace/{namespace}/`

### PR DESCRIPTION
**Pull request by samredai**
### Summary

This adds a minimal amount of detail to the response of the `GET /namespace/{namespace}/` endpoint (name, display_name, type, status, mode, updated_at). See issue #729 for some of the reasoning around this.

Now when going to the page for a namespace in the UI, it loads significantly faster since it only needs to make a single request that runs a single query to get just enough information to display the table (as opposed to a single request per node).

https://github.com/DataJunction/dj/assets/43911210/0dce33ec-9416-4e52-8403-5e78b007d89c

### Test Plan

- [x] PR has an associated issue: #729 
- [x] `make check` passes
- [ ] `make test` shows 100% unit test coverage

### CODE REVIEW

**Comment by shangyian** [comment]
Awesome! We can eventually switch to GraphQL, but this will keep things performant for now

File: datajunction-server/datajunction_server/api/namespaces.py [hunk 1]
```diff
@@ -61,23 +66,25 @@ def create_node_namespace(
...
 @router.get(
     "/namespaces/",
-    response_model=List[str],
+    response_model=List[NamespaceOutput],
     status_code=200,
 )
 def list_namespaces(
     session: Session = Depends(get_session),
-) -> List[str]:
+) -> List[NamespaceOutput]:
```
Comment (shangyian) [inline datajunction-server/datajunction_server/api/namespaces.py:74]: Will this cause the Python client to fail? I think at the moment it assumes that this endpoint returns a list of strings

Comment (samredai) [reply datajunction-server/datajunction_server/api/namespaces.py:74]: Yep, fixed! I just parse out the names right after the call.

Comment (samredai) [reply datajunction-server/datajunction_server/api/namespaces.py:74]: In the future we might want to use the full payload to generate an array of objects or maybe even just a single NodeList object that prints as a nice looking table. 🤔

**Pull request merged**